cloud/parameterstore/parameter_manager.go

```diff
@@ -0,0 +1,196 @@
+	// Parameter Store.
+
+	fullNames := make([]string, 0, len(names))
+	for _, name := range names {
+		fullNames = append(fullNames, pm.getPrefixedName(name))
+	}
+
+	ssmParams, err := pm.ssmClient.GetParametersSimple(ctx, &ssm.GetParametersInput{
+		Names:          fullNames,
+		WithDecryption: aws.Bool(true),
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "getting %d parameters", len(names))
+	}
+
+	params := make([]Parameter, 0, len(ssmParams))
+	for _, p := range ssmParams {
+		name := aws.ToString(p.Name)
+		params = append(params, Parameter{
+			Name:     name,
+			Basename: pm.getBasename(name),
+			Value:    aws.ToString(p.Value),
+		})
+	}
+
+	return params, nil
+}
+
+// GetStrict is the same as Get but verifies that all the requested parameter
+// names were found before returning the result.
+func (pm *ParameterManager) GetStrict(ctx context.Context, names ...string) ([]Parameter, error) {
+	fullNames := make([]string, 0, len(names))
+	for _, name := range names {
+		fullNames = append(fullNames, pm.getPrefixedName(name))
+	}
+
+	params, err := pm.Get(ctx, names...)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(params) != len(fullNames) {
+		foundNames := make(map[string]struct{}, len(params))
+		for _, p := range params {
+			foundNames[p.Name] = struct{}{}
+		}
+
+		var missingNames []string
+		for _, name := range fullNames {
+			if _, ok := foundNames[name]; !ok {
+				missingNames = append(missingNames, name)
+			}
+		}
+
+		if len(missingNames) > 0 {
+			return nil, errors.Errorf("parameter(s) not found: %s", missingNames)
+		}
+	}
+
+	return params, nil
+}
+
+// Delete deletes the parameters given by the provided name(s).
+func (pm *ParameterManager) Delete(ctx context.Context, names ...string) error {
+	if len(names) == 0 {
+		return nil
+	}
+
+	fullNames := make([]string, 0, len(names))
+	for _, name := range names {
+		fullNames = append(fullNames, pm.getPrefixedName(name))
+	}
+
+	_, err := pm.ssmClient.DeleteParameters(ctx, &ssm.DeleteParametersInput{
+		Names: fullNames,
+	})
+	if err != nil {
+		return errors.Wrapf(err, "deleting %d parameters", len(names))
+	}
+
+	// TODO (DEVPROD-9403): update cache as needed.
+
+	return nil
+}
+
+// getPrefixedName returns the parameter name with the common parameter prefix
+// to ensure it is a full path rather than a basename.
+func (pm *ParameterManager) getPrefixedName(basename string) string {
+	if pm.pathPrefix == "" {
+		return basename
+	}
+	if strings.HasPrefix(basename, pm.pathPrefix) {
+		return basename
+	}
+	pathPrefix := strings.TrimSuffix(strings.TrimPrefix(pm.pathPrefix, "/"), "/")
+	return fmt.Sprintf("/%s/%s", pathPrefix, strings.TrimPrefix(basename, "/"))
+}
+
+// getBasename returns the parameter basename without any intermediate paths.
+func (pm *ParameterManager) getBasename(name string) string {
+	idx := strings.LastIndex(name, "/")
+	if idx == -1 {
+		return name
+	}
+	return name[idx+1:]
+}
```

Comment:
can we use [path.Base()](https://pkg.go.dev/path#Base) instead here?

Reply:
I hesitate to do that because path.Base is supposed to be used for URL paths, per the docs:
> The path package should only be used for paths separated by forward slashes, such as the paths in URLs.

It also has behavior that's rather specific to URLs and would be wrong for Parameter Store (like returning "." for an empty string input). Since this slash-delimited scheme is a custom thing specific to Parameter Store and isn't the same as URLs, I think having a custom and tiny helper function to handle Parameter Store's format is fine.